src/components/button/index.js

```diff
@@ -1,107 +1,151 @@
-      buttonClasses.push(styles.outline)
+      this.state = { hover: false }
+      this.onMouseOver = this.onMouseOver.bind(this)
+      this.onMouseOut = this.onMouseOut.bind(this)
    }
 
-   if (outlineNegative) {
-      buttonClasses.push(styles.outlineNegative)
+   onMouseOver() {
+      this.setState({ hover: true })
    }
 
-   if (plain) {
-      buttonClasses.push(styles.plain)
+   onMouseOut() {
+      this.setState({ hover: false })
    }
 
-   if (textClassName) {
-      buttonTextClasses.push(textClassName)
-   }
+   render() {
+      let modeIcon
 
-   // If Buttons are part of a Button Group
-   if (context.buttonGroupStyle) {
-      const buttonGroup = context.buttonGroupStyle
-      style.marginBottom = buttonGroup.spacing
-      style.verticalAlign = 'middle'
+      // buttonColorRGB is an Object with r,g,b values
+      // Sometimes you want to use the color as is directly,
+      // other times when you want to control the Alpha Transparency,
+      // you wan to yse the RGBA values
+      let buttonColor = this.context.OIOStyles.primaryColor
+      let buttonColorRGB = convertColor(buttonColor)
 
-      if (buttonGroup.align === 'left') {
-         style.marginRight = buttonGroup.spacing
-      } else if (buttonGroup.align === 'right') {
-         style.marginLeft = buttonGroup.spacing
+      const buttonClasses = [this.props.className]
+      const buttonTextClasses = [style.text]
+      const buttonName = this.props.name
+
+      const buttonStyle = {
+         backgroundColor: buttonColor,
+         color: '#fff'
       }
-   }
 
-   return (
-      <button className={classNames(buttonClasses)} onClick={onClick} style={style} type={type}>
-         <Icon className={styles.icon} name={icon} />
-         <span className={classNames(styles.text, buttonTextClasses)}>{buttonName}</span>
-         {modeIcon}
-      </button>
-   )
-}
+      buttonClasses.push(style[this.props.size])
 
-Button.propTypes = {
-   className: React.PropTypes.string,
-   icon: React.PropTypes.string,
-   mode: React.PropTypes.string,
-   name: React.PropTypes.string,
-   onClick: React.PropTypes.func,
-   outline: React.PropTypes.bool,
-   outlineNegative: React.PropTypes.bool,
-   plain: React.PropTypes.bool,
-   rounded: React.PropTypes.bool,
-   size: React.PropTypes.string,
-   textClassName: React.PropTypes.string,
-   type: React.PropTypes.string
-}
+      // Button Color by default will use the OIO primary color
+      // Otherwise, it will use the color passed directly to the button
+      if (this.props.color) {
+         buttonColor = this.props.color
+         buttonColorRGB = convertColor(buttonColor)
+      }
 
-Button.defaultProps = {
-   size: 'medium',
-   type: 'button'
-}
+      if (this.props.icon && this.props.name) {
+         buttonClasses.push(style[`${this.props.size}IconAndText`])
+      }
 
-Button.contextTypes = {
-   buttonGroupStyle: React.PropTypes.object
-}
+      if (this.props.icon && !this.props.name) {
```

Comment:
Is there a reason this isn't `else if`? If not, it might be a good idea to discuss conventions around the use of `if`/`else if`